src/actions/index.js

```diff
@@ -0,0 +1,16 @@
+import axios from 'axios';
+
+export const SEARCH_PLAYER = "SEARCH_PLAYER";
+export const SEARCH_PLAYER_SUCCESS = "SEARCH_PLAYER_SUCCESS";
+export const SEARCH_PLAYER_FAIL = "SEARCH_PLAYER_FAIL";
+
+export const searchPlayer = (player) => (dispatch) => {
+    return axios.get(`https://www.balldontlie.io/api/v1/players`)
+                .then((res) =>
+                    dispatch({type: SEARCH_PLAYER_SUCCESS, payload: res.data}))
+                .catch((err) => dispatch({type: SEARCH_PLAYER_FAIL, payload: err}));
+        {
+    //     type: SEARCH_PLAYER,
+    //     payload: player
+    }
```

Comment:
So in hear, it looks like the general patter you were going for was `ACTION_START`, `ACTION_SUCCESS`, `ACTION_FAILURE` pattern where success and failure get triggered by the return value of the http request. You have success and failure set up correctly and they should work they way you expect them to but you have this `SEARCH_PLAYER` action type that's just kind of hanging out and I'm gonna go ahead and assume you meant for it to be `SEARCH_PLAYER_START` and didn't know wheat to do with it. You  would want to put it before your http request gets fire off so that it can switch any sort of fetching data flags you have in sate respectively.